### PR TITLE
fix: table config whith columns that arent in the results

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #2983 
Closes: #3039 
Closes: #3055 

### Description:

Make sure we can render the column names even when they aren't present in the results.

### CODE REVIEW

**Comment by rephus** [comment]
E2E test failed, related ? 
image

**Review by rephus — CHANGES_REQUESTED** [review]
Fix E2E

**Comment by rephus** [comment]
@owlas while you are working this, could you confirm if this change fixes https://github.com/lightdash/lightdash/issues/3039 too ?

**Comment by rephus** [comment]
Sounds like this one is also affected by this bug: https://github.com/lightdash/lightdash/issues/3055 /cc @owlas

**Comment by owlas** [comment]
Yep on it! Think I've got the solution.

Targeting: #2983 #3039 #3055

**Comment by rephus** [comment]
build is failing

File: packages/frontend/src/components/TableConfigPanel/ColumnConfiguration.tsx [hunk 1]
```diff
@@ -10,17 +10,18 @@ export const ColumnConfiguration: React.FC = () => {
...
-            columnOrder,
+            selectedItemIds,
             updateColumnProperty,
             getHeader,
             getDefaultColumnLabel,
             isColumnVisible,
         },
     } = useVisualizationContext();
     const pivotDimension = pivotDimensions?.[0];
+    console.log(selectedItemIds);
```
Comment (rephus) [inline packages/frontend/src/components/TableConfigPanel/ColumnConfiguration.tsx:21]: Leftover console log

**Review by rephus — COMMENTED** [review]
I tested it and everything looks alright . Nice job !

I didn't notice any performance issues. 

Once you fix the build you can merge

**Comment by rephus** [comment]
I noticed a bug: column names are empty on table config
image

**Comment by owlas** [comment]
I managed to reproduce all 3 bugs above on main. But they're all fixed in this branch

**Comment by rephus** [comment]
Pivoted tables does not use  the new table header format... 
image

I'm happy to tackle this on a separate PR

**Review by rephus — APPROVED** [review]
Everything looks good, I can't find any bugs and this is solving the 3 open tickets , so I'm happy to merge this.

**Comment by owlas** [comment]
Great! Will wait for e2e

**Pull request merged**